packages/aws_signature_v4/lib/src/configuration/services/s3.dart

```diff
@@ -73,14 +77,18 @@ class S3ServiceConfiguration extends BaseServiceConfiguration {
     this.chunked = false,
     int chunkSize = _defaultChunkSize,
     this.encoding = S3PayloadEncoding.none,
-  })  : chunkSize = max(chunkSize, _minChunkSize),
+  })  : assert(
+          signPayload || !chunked,
+          'S3 does not accept unsigned, chunked payloads',
```

Comment:
👍 nice to save a network round trip

nit: Would `!signedPayload && chunked` be clearer?

Reply:
That's the check but we need to assert the converse is true